react/I18n/withLocales.jsx

```diff
@@ -1,13 +1,28 @@
 import React from 'react'
 import { I18n, translate } from './'
 
-const withLocales = locales => Component =>
-  translate()(props => {
-    return (
-      <I18n dictRequire={localeCode => locales[localeCode]} lang={props.lang}>
-        <Component {...props} />
-      </I18n>
-    )
-  })
+const withLocales = locales => Component => {
+  // The inner components needs to receive t
+  const Translated = translate()(Component)
+
+  class Wrapped extends React.Component {
+    render() {
+      // Destructure t so that it is not passed downwards
+      // eslint-disable-next-line no-unused-vars
```

Comment:
It would be better to use lodash.omit to avoid this kind of "hack"?

Reply:
Done, thx